transaction.go

```diff
@@ -0,0 +1,71 @@
+package gomatrixserverlib
+
+import (
+	"encoding/json"
+)
+
+// A Transaction is used to push data from one matrix server to another matrix
+// server.
+type Transaction struct {
+	// The ID of the transaction.
+	TransactionID TransactionID `json:"transaction_id"`
+	// The server that sent the transaction.
+	Origin ServerName `json:"origin"`
+	// The server that should receive the transaction.
+	Destination ServerName `json:"destination"`
+	// The millisecond posix timestamp on the origin server when the transction
+	// was created.
+	OriginServerTS Timestamp `json:"origin_server_ts"`
+	// The IDs of the most recent transactions sent by the server.
+	// Mutliple transactions can be sent in parallel so there may be
+	// more than one previous transaction.
+	PreviousIDs []TransactionID `json:"previous_ids"`
+	// The room events pushed by this transaction.
```

Comment:
...what does "pushed" mean in this context? The transaction contains information, and this is it right? I would just say "The XXXX events in this transaction". Ditto `:25`.